xpk.py

```diff
@@ -3169,27 +3519,55 @@ def check_if_workload_can_schedule(args, system):
     xpk_print(f'No ConfigMap exist for cluster with the name {resources_configmap_name}.')
     return True
 
-  device_type = args.tpu_type if args.tpu_type else args.device_type
+  # Check for gke accelerator type:
+  missing_gke_accelerator_type = False
+  if system.gke_accelerator not in cluster_config_map:
+    xpk_print(f'Gke Accelerator Type Check: {args.workload} is requesting {system.gke_accelerator} but '
+    f'cluster only contains {cluster_config_map.keys()}. ')
+    missing_gke_accelerator_type = True
+  elif cluster_config_map[system.gke_accelerator] == _AUTOPROVISIONING_CONFIG_VALUE:
+    # Run total chip check when in autoprovisioning mode.
+    max_chips_in_cluster = int(cluster_config_map[_AUTOPROVISIONING_CONFIG_MAXIMUM_KEY])
+    if system.accelerator_type == AcceleratorType['GPU']:
+      num_chips_in_workload = system.vms_per_slice * system.chips_per_vm * args.num_nodes
+    else:
+      num_chips_in_workload = system.vms_per_slice * system.chips_per_vm * args.num_slices
+
+    if max_chips_in_cluster > num_chips_in_workload:
```

Comment:
Why can't we schedule the workload requesting `num_chips_in_workload` which is less than equal to `max_chips_in_cluster`?

Reply:
Oh no I flipped the if statement and havent tested this code yet. Fixed it. Thanks for the catch!